padertorch/train/hooks.py

```diff
@@ -812,105 +816,115 @@ class StopTraining(Exception):
     pass
 
 
-class LossWeightAnnealingHook(TriggeredHook):
-    """
-    Anneals a loss weight within the los_weights dict of the trainer.
-    """
-    def __init__(self, name, factor, trigger, max_value=None, min_value=None):
+class AnnealingHook(TriggeredHook):
+    def __init__(self, trigger, breakpoints, name):
         """
+        Base class for piece-wise linear annealing. The piece-wise linear
+        function is parameterized by its breakpoints. Can also be used for
+        arbitrary annealing functions by stating breakpoints with an interval
+        similar to the trigger interval.
 
         Args:
-            name: key of the loss_weight
-            factor: factor by which to anneal the loss weight.
-                factor > 1. results in an increase while factor < 1. results
-                in a decrease
             trigger:
-            max_value: upper bound of the weight
-            min_value: lower bound of the weight
-                (hint: can also be used to activate a loss weight after a
-                certain number of iterations/epochs)
+            breakpoints: list of (iteration, value) coordinates of the piecewise linear funtion.
+            name: name of the attribute. You can use "attr1.attr11" to
+                anneal a sub attribute
         """
         super().__init__(trigger)
+        self.breakpoints = sorted(breakpoints, key=lambda x: x[0])
         self.name = name
-        self.factor = factor
-        self.max_value = max_value
-        self.min_value = min_value
+        self.last_break = None
 
     @property
     def uid(self):
-        return super().uid + f"({self.name})"
+        return super().uid + f"({'.'.join(self.name)})"
```

Comment:
Why do you use join here? Name should be already a string.

Reply:
fixed